sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/grpc/ToProtoValueCodecs.java

```diff
@@ -384,5 +499,126 @@ public UUIDCodec(String typeDesc) {
+        return invalidStringValue(value);
+      } catch (UnknownHostException e) {
+        return invalidStringValue(value);
+      }
+    }
+  }
+
+  protected static final class TimestampCodec extends ToProtoCodecBase {
+    public TimestampCodec() {
+      super("TypeSpec.Basic.TIMESTAMP");
+    }
+
+    @Override
+    public QueryOuterClass.Value protoValueFromStrictlyTyped(Object value) {
+      if (value instanceof String) {
+        return protoValueFromStringified((String) value);
+      }
+      return cannotCoerce(value);
+    }
+
+    @Override
+    public QueryOuterClass.Value protoValueFromStringified(String value) {
+      try {
+        // TODO: this implementation requires full date/time specification including timezone
+        // we could support more flexibility in format as requested in
+        // https://github.com/stargate/stargate/issues/839
+        return Values.of(Instant.parse(value).toEpochMilli());
+      } catch (IllegalArgumentException e) {
+        return invalidStringValue(value);
+      }
+    }
+  }
+
+  protected static final class DateCodec extends ToProtoCodecBase {
+    public DateCodec() {
+      super("TypeSpec.Basic.DATE");
+    }
+
+    @Override
+    public QueryOuterClass.Value protoValueFromStrictlyTyped(Object value) {
+      if (value instanceof String) {
+        return protoValueFromStringified((String) value);
+      }
+      return cannotCoerce(value);
+    }
+
+    @Override
+    public QueryOuterClass.Value protoValueFromStringified(String value) {
+      try {
+        return Values.of(LocalDate.parse(value));
+      } catch (IllegalArgumentException e) {
+        return invalidStringValue(value);
+      }
+    }
+  }
+
+  protected static final class TimeCodec extends ToProtoCodecBase {
+    public TimeCodec() {
+      super("TypeSpec.Basic.TIME");
+    }
+
+    @Override
+    public QueryOuterClass.Value protoValueFromStrictlyTyped(Object value) {
+      if (value instanceof String) {
+        return protoValueFromStringified((String) value);
+      }
+      return cannotCoerce(value);
+    }
+
+    @Override
+    public QueryOuterClass.Value protoValueFromStringified(String value) {
+      try {
+        return Values.of(LocalTime.parse(value));
+      } catch (IllegalArgumentException e) {
+        return invalidStringValue(value);
+      }
+    }
+  }
+
+  protected static final class BlobCodec extends ToProtoCodecBase {
+
+    public BlobCodec() {
+      super("TypeSpec.Basic.BLOB");
+    }
+
+    @Override
+    public QueryOuterClass.Value protoValueFromStrictlyTyped(Object value) {
+      if (value instanceof byte[]) {
+        return Values.of((byte[]) value);
+      } else if (value instanceof ByteBuffer) {
+        return Values.of((ByteBuffer) value);
+      }
+      return cannotCoerce(value);
+    }
+
+    @Override
+    public QueryOuterClass.Value protoValueFromStringified(String value) {
+      return Values.of(value.getBytes());
```

Comment:
Hmmmh. If String, would have to be base64-encoded. So I suspect this is wrong.

Reply:
From SGv1 `Converters`:

```
Base64.getDecoder().decode(value)
```

is what is needed I think. May throw exception for invalid base64 encoding.

Reply:
Suggest we should use `io.stargate.core.util.ByteBufferUtils' for inbound/outbound conversions, sound correct?

Reply:
If they are general purpose yes (no proto/grpc type dependencies). Assuming this would be in sgv2-service-common?

Reply:
This is currently in `core` but does not appear to be used in any coordinator modules on the backend. We might be able to move to `sgv2-service-common`, but I think we can defer that decision for now.